### PR TITLE
Thoth.Json migration

### LINKED ISSUE #312: [Feature Request] Update Thoth.Json
Might bring potential performance gain

Nuget:
[Thoth.Json.Core](https://www.nuget.org/packages/Thoth.Json.Core)
[Thoth.Json.Newtonsoft](https://www.nuget.org/packages/Thoth.Json.Newtonsoft)
[Thoth.Json.Newtonsoft](https://www.nuget.org/packages/Thoth.Json.Newtonsoft)

### PR DESCRIPTION
**Pull request by HLWeil**
closes #312 

- [x] Tests needed for Value Conversion (Parse float with different culture settings) (nvm this is already being tested)

### CODE REVIEW

File: src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj [hunk 1]
```diff
@@ -40,8 +40,9 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="NJsonSchema" Version="10.8.0" />
-    <PackageReference Include="Thoth.Json" Version="10.1.0" />
-    <PackageReference Include="Thoth.Json.Net" Version="11.0.0" />
+    <PackageReference Include="Thoth.Json.Core" Version="0.2.1" />
+    <PackageReference Include="Thoth.Json.JavaScript" Version="0.1.0" />
```
Comment (Freymaurer) [inline src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj:44]: No python package? I have not looked into this, just wondering

File: src/ISA/ISA.Json/Decode.fs [hunk 2]
```diff
@@ -1,49 +1,63 @@
...
 open ARCtrl.ISA
 open Fable.Core
 
 module GDecode =
 
+    let helpers = 
+        #if FABLE_COMPILER
+        Thoth.Json.JavaScript.Decode.helpers
+        #else
+        Thoth.Json.Newtonsoft.Decode.helpers
```
Comment (Freymaurer) [inline src/ISA/ISA.Json/Decode.fs:14]: Again, no python necessary?

File: src/ISA/ISA.Json/Ontology.fs [hunk 3]
```diff
@@ -17,18 +14,21 @@ module AnnotationValue =
...
-                    | Error e -> Error e
+                    match Decode.float.Decode(s,json) with
+                    | Ok f -> Ok <| string f
+                    | Error _ -> 
+                        match Decode.string.Decode(s,json) with
+                        | Ok s -> Ok <| s
+                        | Error e -> Error e       
+        }
 
+            // is there a option to decode force to string?
```
Comment (Freymaurer) [inline src/ISA/ISA.Json/Ontology.fs:30]: this todo is solved?

File: src/ISA/ISA.Json/Publication.fs [hunk 4]
```diff
@@ -19,22 +16,24 @@ module Publication =
                             | Some t -> "#Pub_" + t.Replace(" ","_")
                             | None -> "#EmptyPublication"
 
-    let rec encoder (options : ConverterOptions) (oa : obj) = 
+    let rec encoder (options : ConverterOptions) (oa : Publication) =
```
Comment (Freymaurer) [inline src/ISA/ISA.Json/Publication.fs:19]: why "rec"?

File: src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj (hunk 1 above)
Comment (HLWeil) [reply src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj:44]: This is only the thoth update, without any python integration. The will come in the next, separate PR.

File: src/ISA/ISA.Json/Decode.fs (hunk 2 above)
Comment (HLWeil) [reply src/ISA/ISA.Json/Decode.fs:14]: Same

File: src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj (hunk 1 above)
Comment (Freymaurer) [reply src/ISA/ISA.Json/ARCtrl.ISA.Json.fsproj:44]: Ohhhh!

File: src/ISA/ISA.Json/Ontology.fs (hunk 3 above)
Comment (HLWeil) [reply src/ISA/ISA.Json/Ontology.fs:30]: Nope, could open up an issue at Fable, but not sure how you would even call such a function.

File: src/ISA/ISA.Json/Publication.fs (hunk 4 above)
Comment (HLWeil) [reply src/ISA/ISA.Json/Publication.fs:19]: Probably a copy error. Removed now.

**Review by Freymaurer — APPROVED** [review]
lgtm 🚀

**Pull request merged**